src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -381,15 +389,38 @@ protected boolean doEquals(NeuralQueryBuilder obj) {
         EqualsBuilder equalsBuilder = new EqualsBuilder();
         equalsBuilder.append(fieldName, obj.fieldName);
         equalsBuilder.append(queryText, obj.queryText);
+        equalsBuilder.append(queryImage, obj.queryImage);
         equalsBuilder.append(modelId, obj.modelId);
         equalsBuilder.append(k, obj.k);
+        equalsBuilder.append(maxDistance, obj.maxDistance);
+        equalsBuilder.append(minScore, obj.minScore);
+        equalsBuilder.append(expandNested, obj.expandNested);
+        equalsBuilder.append(getVector(vectorSupplier), getVector(obj.vectorSupplier));
         equalsBuilder.append(filter, obj.filter);
+        equalsBuilder.append(methodParameters, obj.methodParameters);
+        equalsBuilder.append(rescoreContext, obj.rescoreContext);
         return equalsBuilder.isEquals();
     }
 
     @Override
     protected int doHashCode() {
-        return new HashCodeBuilder().append(fieldName).append(queryText).append(modelId).append(k).toHashCode();
+        return new HashCodeBuilder().append(fieldName)
```

Comment:
Please use `Objects.hash` method, it's widely used in other OS repos, you can check some example of usage here:  https://github.com/opensearch-project/OpenSearch/blob/845fbfa10407f3264e6aab8812eff4ef0ad8be24/server/src/main/java/org/opensearch/search/aggregations/metrics/TopHitsAggregationBuilder.java#L919-L936